modinfo/src/hints.rs

```diff
@@ -0,0 +1,100 @@
+use std::collections::HashMap;
+
+use game::HintGhost;
+use serde::{ser::SerializeMap, Serialize};
+
+use crate::text::Text;
+
+#[derive(Debug, Default)]
+pub struct Hints {
+    pub ghosts: HashMap<HintGhost, Text<'static>>,
```

Comment:
Collapsing the different Hint types into a single map causes them to lose information in the spoiler log about what type of Hint they were. I think that's valuable information worth preserving.

Reply:
Personally I disagree, but I am curious if you got this idea from another randomizer. I'm not personally familiar with any other randomizer that organizes hints in the spoiler log this way.

Reply:
I haven't looked too closely at other rando's spoiler logs, I just wanted ours to be, well, useful. Organization helps with that IMO, but we may just have to agree to disagree here.

Don't remember if I committed this or not yet, but I made a change recently to have Path Hints actually list the underlying Path Item in the spoiler log. That kind of info is really great to help people understand the hint, not to mention how helpful it is for debugging.